src/model/immutable/EntityMap.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule EntityMap
+ * @flow
+ */
+
+'use strict';
+
+import type DraftEntityInstance from 'DraftEntityInstance';
+import type {OrderedMap} from 'immutable';
+
+export type EntityMap = OrderedMap<string, DraftEntityInstance>;
```

Comment:
I chose an `OrderedMap` because of the fact that we need to know the last inserted key. I was under the impression that only an `OrderedMap` would be safe for that, since the order of iteration for `Map` is supposed to not be defined. If we can be sure that the last inserted key is always the actually last inserted key, then sure, we can use `Map` here.